API/src/controlers/parcelControllers.js

```diff
@@ -0,0 +1,96 @@
+import Parcel from '../models/parcel';
+
+const controllers = {};
+// declare the variable to store orders.
+const orders = [];
+
+// fetch a parcel
+const fetchParcelById = (req, res) => {
+  const parcelId = parseInt(req.params.id);
+  const parcel = orders.find(order => order.id === parcelId);
+  // send it.
+  if (parcel) {
+    res.status(200).send(parcel);
+  } else {
+    // send the error page
+    res.send({ message: 'Ooops! no order with that id' });
+  }
+};
+
+// create parcel
+const createParcel = (req, res) => {
+  const {
+    name, origin, destination, weight, userId,
+  } = req.body;
+  const existingOrder = orders.find(order => order.name === name);
+  if (existingOrder === undefined) {
+    const id = orders.length + 1;
+    const fieldsValidation = /[a-zA-Z]+/;
+    if (!origin) {
```

Comment:
This block of code is all handling required fields.
We could instead use a generic response message for the required fields.

```
if (!weight || !name || !destination) { 
  res.send({ message: 'Please provide all the required fields!' });
}
```